spec/integration/spec_helper.cr

```
@@ -0,0 +1,172 @@
+    end
+  end
+end
+
+def refute_installed(name, version = nil, file = __FILE__, line = __LINE__)
+  if version
+    if Dir.exists?(install_path(name))
+      assert File.exists?(install_path(name, "shard.yml")), "expected shard.yml for installed #{name} dependency was not found", file, line
+      spec = Shards::Spec.from_file(install_path(name, "shard.yml"))
+      spec.version.should_not eq(version), file, line
+    end
+  else
+    refute Dir.exists?(install_path(name)), "expected #{name} dependency to not have been installed", file, line
+  end
+end
+
+def assert_installed_file(path, file = __FILE__, line = __LINE__)
+  assert File.exists?(File.join(install_path(name), path)), "Expected #{path} to have been installed", file, line
+end
+
+def assert_locked(name, version = nil, file = __FILE__, line = __LINE__)
+  path = File.join(application_path, "shard.lock")
+  assert File.exists?(path), "expected shard.lock to have been generated", file, line
+  locks = Shards::Lock.from_file(path)
+  assert lock = locks.find { |d| d.name == name }, "expected #{name} dependency to have been locked", file, line
+
+  if lock && version
+    if version =~ Shards::VERSION_REFERENCE
+      assert version == lock.version, "expected #{name} dependency to have been locked at version #{version}", file, line
+    else
+      assert version == lock.refs, "expected #{name} dependency to have been locked at commit #{version}", file, line
+    end
+  end
+end
+
+def refute_locked(name, version = nil, file = __FILE__, line = __LINE__)
+  path = File.join(application_path, "shard.lock")
+  assert File.exists?(path), "expected shard.lock to have been generated", file, line
+  locks = Shards::Lock.from_file(path)
+  refute locks.find { |d| d.name == name }, "expected #{name} dependency to not have been locked", file, line
+end
+
+def install_path(project, *path_names)
+  File.join(application_path, "lib", project, *path_names)
+end
+
+def debug(command)
```

What is this used for?

---

I guess it's a helper method to debug failure on commands. It was there I didn't want to disrupt anybody's workflow.

---

Yep. Useful when an integration test starts failing: add a debug run instead of trying to reproduce the test case or printing the output manually.